comp/rdnsquerier/impl/rdnsquerier.go

```diff
@@ -48,16 +78,83 @@ func NewComponent(reqs Requires) (Provides, error) {
 	}, nil
 }
 
+// GetHostname attempts to use reverse DNS lookup to resolve the hostname for the given IP address.
+// If the IP address is in the private address space and the lookup is successful then the updateHostname
+// function will be called with the hostname.
+func (q *rdnsQuerierImpl) GetHostname(ipAddr []byte, updateHostname func(string)) {
+	ipaddr, ok := netip.AddrFromSlice(ipAddr)
+	if !ok {
+		// IP address is invalid
+		return
```

Comment:
It would be nice to get some feedback when the addr was malformed, either by logging here or by returning an error from `GetHostname` so that the calling program can log it

Reply:
I plan to add telemetry here and in other cases where we're unable to successfully perform the rDNS lookup, and for now I added a trace log.  In the netflow use case I don't think we care about handling an invalid IP address error - if a flow has invalid IP address(es) we just won't be able to enrich it.